cmds/exp/bzimage/bzimage.go

```diff
@@ -57,143 +59,144 @@ var (
 	jsonOut = flag.BoolP("json", "j", false, "json output ('ver' subcommand only)")
 )
 
-func usage() {
-	fmt.Fprintln(os.Stderr, cmdUsage)
-	flag.PrintDefaults()
-	os.Exit(1)
-}
-
-func main() {
-	flag.Usage = usage
-	flag.Parse()
-
+func run(w io.Writer, args ...string) error {
 	if *debug {
 		bzimage.Debug = log.Printf
 	}
-	a := flag.Args()
-	if len(a) < 2 {
-		usage()
+	if len(args) < 2 {
+		flag.Usage()
```

Comment:
could we return an error that is the usage message?

Reply:
No in my opinion not. In i an error is returned it will log.Fatal(err) the error in main. So the usage will not printed properly. Do you have any suggestions?

Reply:
Do you want this feature for testing purpose? My suggestion in that case would be to print that with the help of an io.Writer to StdErr. Like this we can write it to a buffer to test it. How about that?